src/configuration.ts

```diff
@@ -17,7 +17,7 @@ export const initializeStorageLocation = (location: string) => {
         }
         const extensionFilePath = getAnnotationFilePath();
         if (!fs.existsSync(extensionFilePath)) {
-            fs.writeFileSync(extensionFilePath, '{"notes":[], "nextId":1}');
+            fs.writeFileSync(extensionFilePath, '{"notes":[], "nextId":1, "customTODO": []}');
```

Comment:
I think the best way to do this is not store this setting on the "database", but instead have it configured as a seeting on `package.json`, and to modify the information, users can access the configuration of the extension and add/remove things there.

For reference, check how [project manager](https://github.com/alefragnani/vscode-project-manager/blob/9f8677aa3d6150b5fdb401fe622403167e82c9bc/package.json) handles the `projectManager.vscode.ignoredFolders` option.

By using this, you can modify the [configuration](https://github.com/thamara/vscode-code-annotation/blob/ebf5fd3fb98d34de528c7803654842d7be60ae71/src/configuration.ts#L27), to include this new arraay of strings (representing the regexps), and load it into the configuration in ` getConfiguration` (in the same file: `configuration.ts`.

The matching code should be kept pretty much what you doing, with the difference that you'll be sourcing the array from `getConfiguration`.

Reply:
I've changed it to use the configuration instead and removed the commands which were to configure them. I've also added a bit of error handling to the check in case of invalid regexp and removed a couple of unused imports.

Reply:
Thanks! I'll get to review this later today, but looks really good!